plugins/polio/api.py

```diff
@@ -2082,6 +2091,230 @@ def get_queryset(self):
+    def has_permission(self, request, view):
+        read_perm = permission.POLIO_VACCINE_AUTHORIZATIONS_READ_ONLY
+        write_perm = permission.POLIO_VACCINE_AUTHORIZATIONS_ADMIN
+        if request.method == "GET":
+            can_get = (
+                request.user
+                and request.user.is_authenticated
+                and request.user.has_perm(read_perm)
+                or request.user.is_superuser
+            )
+            return can_get
+        elif (
+            request.method == "POST"
+            or request.method == "PUT"
+            or request.method == "PATCH"
+            or request.method == "DELETE"
+        ):
+            can_post = (
+                request.user
+                and request.user.is_authenticated
+                and request.user.has_perm(write_perm)
+                or request.user.is_superuser
+            )
+            return can_post
+        else:
+            return False
+
+
+def handle_none_and_country(item, ordering):
+    """
+    This function handle the None cases to order the response of get_most_recent_authorizations
+    and country nested dict.
+    """
+
+    if "country" in ordering:
+        country_dict = item.get("country")
+        country_name = country_dict.get("name")
+        return country_name if country_name is not None else ""
+
+    if "date" in ordering:
+        item = item.get(ordering)
+        return item if item is not None else dt.date(1, 1, 1)
+
+    if "date" not in ordering:
+        item = item.get(ordering)
+        return item if item is not None else float("inf")
+
+
+@swagger_auto_schema(tags=["vaccineauthorizations"])
+class VaccineAuthorizationViewSet(ModelViewSet):
+    """
+    Vaccine Authorizations API
+    list: /api/polio/vaccintauthorizations
+    action: /api/polio/get_most_recent_authorizations
+    """
+
+    permission_classes = [HassVaccineAuthorizationsPermissions]
+    filter_backends = [filters.OrderingFilter, DjangoFilterBackend, DeletionFilterBackend]
+    results_key = "results"
+    remove_results_key_if_paginated = True
+    serializer_class = VaccineAuthorizationSerializer
+    pagination_class = Paginator
+    ordering_fields = ["status", "current_expiration_date", "next_expiration_date", "expiration_date", "quantity"]
+
+    def get_queryset(self):
+        user = self.request.user
+        user_access_ou = OrgUnit.objects.filter_for_user_and_app_id(user, None)
+        user_access_ou = user_access_ou.filter(org_unit_type__name="COUNTRY")
+        country_id = self.request.query_params.get("country", None)
+        queryset = VaccineAuthorization.objects.filter(account=user.iaso_profile.account, country__in=user_access_ou)
+        block_country = self.request.query_params.get("block_country", None)
+        search = self.request.query_params.get("search", None)
+        auth_status = self.request.query_params.get("auth_status", None)
+
+        if country_id:
```

Comment:
I much prefer to define or extend custom filters for this, block_country ... search .. auth_status .. See [iaso/api/deduplication/filters.py](https://github.com/BLSQ/iaso/blob/ccc24e3e2ca3edd351b923aaaefe43e13a0e418c/iaso/api/deduplication/filters.py) and `EntityDuplicateViewSet` in [iaso/api/deduplication/entity_duplicate.py](https://github.com/BLSQ/iaso/blob/fb93ea9bd51cf86bbde48a10c1a9a943249b96e7/iaso/api/deduplication/entity_duplicate.py) for some examples ... It might be personal preferences but it's much cleaner to my taste